File: backend/src/main/java/org/vaccom/vcmgt/entity/CoSoYTe.java
package org.vaccom.vcmgt.entity;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import org.vaccom.vcmgt.constant.DBConstant;

@Entity
@Table(name = DBConstant._COSOYTE_TABLE_NAME)
public class CoSoYTe implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	public CoSoYTe() {

	}

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "ID")
	private long id;

	@Column(name = "MaCoSo")
	private String maCoSo;

	@Column(name = "TenCoSo")
	private String tenCoSo;

	@Column(name = "TinhThanhMa")
	private String tinhThanhMa;

	@Column(name = "TinhThanhTen")
	private String tinhThanhTen;

	@Column(name = "QuanHuyenMa")
	private String quanHuyenMa;

	@Column(name = "QuanHuyenTen")
	private String quanHuyenTen;

	@Column(name = "PhuongXaMa")
	private String phuongXaMa;

	@Column(name = "PhuongXaTen")
	private String phuongXaTen;

	@Column(name = "DiaChiCoSo")
	private String diaChiCoSo;

	@Column(name = "NguoiDaiDien")
	private String nguoiDaiDien;

	@Column(name = "SoDienThoai")
	private String soDienThoai;

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public String getMaCoSo() {
		return maCoSo;
	}

	public void setMaCoSo(String maCoSo) {
		this.maCoSo = maCoSo;
	}

	public String getTenCoSo() {
		return tenCoSo;
	}

	public void setTenCoSo(String tenCoSo) {
		this.tenCoSo = tenCoSo;
	}

	public String getTinhThanhMa() {
		return tinhThanhMa;
	}

	public void setTinhThanhMa(String tinhThanhMa) {
		this.tinhThanhMa = tinhThanhMa;
	}

	public String getTinhThanhTen() {
		return tinhThanhTen;
	}

	public void setTinhThanhTen(String tinhThanhTen) {
		this.tinhThanhTen = tinhThanhTen;
	}

	public String getQuanHuyenMa() {
		return quanHuyenMa;
	}

	public void setQuanHuyenMa(String quanHuyenMa) {
		this.quanHuyenMa = quanHuyenMa;
	}

	public String getQuanHuyenTen() {
		return quanHuyenTen;
	}

	public void setQuanHuyenTen(String quanHuyenTen) {
		this.quanHuyenTen = quanHuyenTen;
	}

	public String getPhuongXaMa() {
		return phuongXaMa;
	}

	public void setPhuongXaMa(String phuongXaMa) {
		this.phuongXaMa = phuongXaMa;
	}

	public String getPhuongXaTen() {
		return phuongXaTen;
	}

	public void setPhuongXaTen(String phuongXaTen) {
		this.phuongXaTen = phuongXaTen;
	}

	public String getDiaChiCoSo() {
		return diaChiCoSo;
	}

	public void setDiaChiCoSo(String diaChiCoSo) {
		this.diaChiCoSo = diaChiCoSo;
	}

	public String getNguoiDaiDien() {
		return nguoiDaiDien;
	}

	public void setNguoiDaiDien(String nguoiDaiDien) {
		this.nguoiDaiDien = nguoiDaiDien;
	}

	public String getSoDienThoai() {
		return soDienThoai;
	}

	public void setSoDienThoai(String soDienThoai) {
		this.soDienThoai = soDienThoai;
	}

}

File: mobile/Vaccom_Mobile/lib/network/response/mapping/cosoyte.dart
import 'package:vaccom_mobile/model/base_item.dart';

class CoSoYTe implements BaseItem {
  CoSoYTe({
    this.id,
    this.code,
    this.name,
    this.tinhThanhMa,
    this.tinhThanhTen,
    this.quanHuyenMa,
    this.quanHuyenTen,
    this.phuongXaMa,
    this.phuongXaTen,
    this.diaChiCoSo,
    this.nguoiDaiDien,
    this.soDienThoai,
    this.maQr,
  });

  int id, parentID;
  String code;
  String name;
  String tinhThanhMa;
  String tinhThanhTen;
  String quanHuyenMa;
  String quanHuyenTen;
  String phuongXaMa;
  String phuongXaTen;
  String diaChiCoSo;
  String nguoiDaiDien;
  String soDienThoai;
  String maQr;

  factory CoSoYTe.fromJson(Map<String, dynamic> json) => CoSoYTe(
        id: json["id"],
        code: json["maCoSo"],
        name: json["tenCoSo"],
        tinhThanhMa: json["tinhThanhMa"],
        tinhThanhTen: json["tinhThanhTen"],
        quanHuyenMa: json["quanHuyenMa"],
        quanHuyenTen: json["quanHuyenTen"],
        phuongXaMa: json["phuongXaMa"],
        phuongXaTen: json["phuongXaTen"],
        diaChiCoSo: json["diaChiCoSo"],
        nguoiDaiDien: json["nguoiDaiDien"],
        soDienThoai: json["soDienThoai"],
        maQr: json["maQR"],
      );

  Map<String, dynamic> toJson() => {
        "id": id,
        "maCoSo": code,
        "tenCoSo": name,
        "tinhThanhMa": tinhThanhMa,
        "tinhThanhTen": tinhThanhTen,
        "quanHuyenMa": quanHuyenMa,
        "quanHuyenTen": quanHuyenTen,
        "phuongXaMa": phuongXaMa,
        "phuongXaTen": phuongXaTen,
        "diaChiCoSo": diaChiCoSo,
        "nguoiDaiDien": nguoiDaiDien,
        "soDienThoai": soDienThoai,
        "maQR": maQr,
      };

  static List<CoSoYTe> listFromJson(dynamic data) =>
      List<CoSoYTe>.from(data.map((x) => CoSoYTe.fromJson(x)));
}
